src/libraries/System.Private.CoreLib/src/System/Span.cs

```diff
@@ -300,19 +300,7 @@ public unsafe void Clear()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public unsafe void Fill(T value)
         {
-            if (sizeof(T) == 1)
```

Comment:
Is the SpanHelpers byte overload more efficient for sizeof(T) == 1 or there is no difference? If the byte overload is more efficient, we may want to keep this check and call the byte overload for sizeof(T) == 1.

Reply:
> Is the SpanHelpers byte overload more efficient for sizeof(T) == 1 or there is no difference?

No difference for now, but I wanted to improve the byte version so there will be